File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0;
        int right = (int)heights.size() - 1;

        int res = INT_MIN;

        while(left < right) {
            res = max(res, min(heights[left], heights[right]) * (right - left));

            if(heights[left] <= heights[right]) {
                left += 1;
            }
            else {
                right -= 1;
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = 0

        def dfs(node: Optional, max_val: int) -> None:
            nonlocal res

            if not node:
                return None
            
            if(node.val >= max_val):
                res += 1
            
            dfs(node.left, max(max_val, node.val))
            dfs(node.right, max(max_val, node.val))

            return

        dfs(root, float('-inf'))

        return res